src/Language/SystemVerilog/Parser/Preprocess.hs

```diff
@@ -148,6 +158,64 @@ isIdentChar ch =
     ('0' <= ch && ch <= '9') ||
     (ch == '_') || (ch == '$')
 
+-- Update a Cond for an `else block: if we've never had a hit, this is the one
+-- for us. Otherwise a previous block was true, so it isn't.
+elseCond :: Cond -> Cond
+elseCond NeverTrue = CurrentlyTrue
+elseCond _ = NeverTrue
+
+-- Return a Cond for whether a name is defined (assuming that it hasn't been
+-- defined previously).
+ifdefCond :: Env -> String -> Cond
+ifdefCond env name = if Map.member name env then CurrentlyTrue else NeverTrue
+
+-- Update a Cond for an `elsif block. The boolean argument is whether the
+-- `elsif block's test is true.
+elsifCond :: Env -> String -> Cond -> Cond
+elsifCond env name c =
+    case c of
+        NeverTrue -> ifdefCond env name
+        _ -> PreviouslyTrue
+
+-- Push a condition onto the top of the preprocessor condition stack
+pushCondStack :: Cond -> PPS ()
+pushCondStack c = getCondStack >>= (\ cs -> setCondStack (c : cs))
+
+-- Pop the top from the preprocessor condition stack. If the stack is empty,
+-- raise a lexical error based on name (the name of the directive that caused
+-- the problem).
```

Comment:
This file generally uses briefs of the form:
```
-- pop the top from the preprocessor condition stack
```
I don't think it's necessary to mention the lexical error in the brief. The rest of the brief might be obviated if we rename `name` to `directive`.

Reply:
Cool, done.